app/client/src/components/shared/MapWidgets.js

```diff
@@ -636,21 +640,23 @@ function MapWidgets({
 
     Promise.all(requests)
       .then((responses) => {
-        setAdditionalLegendInfo({
+        additionalLegendInfoNonState = {
           status: 'success',
           data: {
             protectedAreasLayer: responses[0],
             ejscreen: responses[1],
             mappedWaterLayer: responses[2],
           },
-        });
+        };
+        setAdditionalLegendInfo(additionalLegendInfoNonState);
       })
       .catch((err) => {
         console.error(err);
-        setAdditionalLegendInfo({
+        additionalLegendInfoNonState = {
           status: 'failure',
           data: {},
-        });
+        };
+        setAdditionalLegendInfo(additionalLegendInfoNonState);
       });
   }, [additioanlLegendInitialized, services]);
```

Comment:
A minor point, but the spelling is a little reversed here. It is consistent, but it could be confusing at some point.
```suggestion
  }, [additionalLegendInitialized, services]);
```

Reply:
Good catch. I'll fix that spelling real quick.